File: campaign_manager/gos/models/spell.py
from django.db import models

class Spell(models.Model):
    created = models.DateTimeField(auto_now_add=True)
    name = models.CharField(max_length=100, default='')
    description = models.TextField()
    level = models.IntegerField()
    casting_time = models.CharField(max_length=20)
    range = models.CharField(max_length=20)
    components = models.CharField(max_length=20, blank=True)
    duration = models.CharField(max_length=30)
    school = models.CharField(max_length=50)
    attack_save = models.CharField(max_length=100)
    damage_effect = models.CharField(max_length=100)
    higher_levels = models.CharField(max_length=255, blank=True)
    
    class Meta:
        ordering = ['name']

File: src/app/class/spell/spell.ts
export class Spell {
    id: number;
    created?: string;
    name: string;
    description?: string;
    level?: string;
    castingTime?: string;
    range?: string;
    components?: string;
    duration?: string;
    school?: string;
    attackSave?: string;
    damageEffect?: string;
    higherLevels?: string;

    constructor(props: any) {
        this.id = props.id;
        this.created = props.created;
        this.name = props.name;
        this.description = props.description;
        this.level = props.level;
        this.castingTime = props.castingTime;
        this.range = props.range;
        this.components = props.components;
        this.duration = props.duration;
        this.school = props.school;
        this.attackSave = props.attackSave;
        this.damageEffect = props.damageEffect;
        this.higherLevels = props.higherLevels;
    }
}
